src/node/rpc/frontend.h

```diff
@@ -589,7 +589,7 @@ namespace ccf
           // If serialising the committed transaction fails, there is no way to
           // recover safely (https://github.com/microsoft/CCF/issues/338).
           // Better to abort.
-          LOG_FATAL_FMT(e.what());
+          throw;
```

Comment:
I had a comment here but it seems GitHub lost it?

Removing the macro means we remove the distinction between errors we know are recoverable (`throw std::logic_error`) and errors we think are really fatal ("I'd like to `abort` but I'll at least try to tell you why, and where"). In most of these instances that looks fine (we may be able to catch these safely, and in general they'll kill the node but in the same way as our other asserts).

I'm not sure about this instance though - if we get here, our kv is half-modified and we really do want to kill this node ASAP? I guess the threat is that another thread responds from the corrupt kv, and we can't reliably intercept that, so maybe this is fine. Thinking out loud.

Reply:
I don't think we do, because we throw the same exception either way. If the logging code has abort()ed, I would have agreed, but as it stands, I think it's really equivalent.

In this situation, we've committed a Tx, and we can't serialise it, there is no way out, so it is simpler to die than to confuse the other nodes?

Reply:
Agreed.